### PR TITLE
Add id to the cancer types result tab

### PR DESCRIPTION
**Pull request by oplantalech**
Add id to the cancer types result tab to allow to disable cancer types result tab when querying for only gene sets. This PR is related to cbioportal/cbioportal-frontend#1008.

### CODE REVIEW

File: portal/src/main/webapp/WEB-INF/jsp/visualize.jsp [hunk 1]
```diff
@@ -495,7 +503,11 @@ window.loadReactApp({ defaultRoute: 'results' });
         );
 
         //Move code related to expression tab from cross_cancer_results.jsp to here
-        window.ccQueriedGenes = OQL.genes(window.serverVars.theQuery);
+        window.ccQueriedGenes;
+        var isGenesetQuery = jspToJs('<%=isGenesetQuery%>');
+        if (!isGenesetQuery) {
+            window.ccQueriedGenes = OQL.genes(window.serverVars.theQuery);
+        }
```
Comment (oplantalech) [inline portal/src/main/webapp/WEB-INF/jsp/visualize.jsp]: @adamabeshouse I haven't introduced this OQL line here, I just added a check so that this line is skipped when having queries with only gene sets. This was added in #3071, which you approved. Maybe you can recheck it and then we decide what to do here.

Comment (adamabeshouse) [reply portal/src/main/webapp/WEB-INF/jsp/visualize.jsp]: like I said, lines 506+ are good. I was talking about this: https://github.com/cBioPortal/cbioportal/pull/3940/files#diff-5c6909f939f5f14fdf689eaac83ae729R78

**Comment by oplantalech** [comment]
@adamabeshouse I've moved the hiding of the tabs to cbioportal/cbioportal-frontend#1008. In the jsp part I only need an id name for the "Cancer Types" tab, that's what I've made here.

File: portal/src/main/webapp/WEB-INF/jsp/visualize.jsp [hunk 2]
```diff
@@ -186,7 +186,7 @@ window.loadReactApp({ defaultRoute: 'results' });
             out.println ("<li><a href='#summary' class='result-tab' id='oncoprint-result-tab'>OncoPrint</a></li>");
             // if showCancerTypesSummary is try, add the list item
             if(showCancerTypesSummary){
-                out.println ("<li><a href='#pancancer_study_summary' class='result-tab' title='Cancer types summary'>"
```
Comment (adamabeshouse) [inline portal/src/main/webapp/WEB-INF/jsp/visualize.jsp]: why did you delete the `title` attribute?

File: portal/src/main/webapp/WEB-INF/jsp/visualize.jsp [hunk 3]
```diff
@@ -495,7 +495,10 @@ window.loadReactApp({ defaultRoute: 'results' });
         );
 
         //Move code related to expression tab from cross_cancer_results.jsp to here
+        window.ccQueriedGenes;
```
Comment (adamabeshouse) [inline portal/src/main/webapp/WEB-INF/jsp/visualize.jsp]: unnecessary line

File: portal/src/main/webapp/WEB-INF/jsp/visualize.jsp [hunk 4]
```diff
@@ -495,7 +495,10 @@ window.loadReactApp({ defaultRoute: 'results' });
...
+        if (!window.serverVars.theQuery) {
```
Comment (adamabeshouse) [inline portal/src/main/webapp/WEB-INF/jsp/visualize.jsp]: this logic seems wrong, did you mean `if(window.serverVars.theQuery){` (no !)

**Comment by oplantalech** [comment]
@adamabeshouse Thanks for the comments! Now all should be fixed.

**Comment by adamabeshouse** [comment]
@oplantalech please squash your commits and rebase onto rc

**Comment by oplantalech** [comment]
@adamabeshouse Done
